include/deal.II/hp/fe_collection.h

```diff
@@ -274,6 +274,45 @@ namespace hp
     unsigned int
     find_least_face_dominating_fe(const std::set<unsigned int> &fes) const;
 
+    /**
+     * Try to find a most face dominating finite element inside the subset of
+     * fe_indices @p fes as part of this FECollection. For example, if an
+     * FECollection consists of `{FE_Q(1),FE_Q(2),FE_Q(3),FE_Q(4)}` elements
+     * and we are looking for the most face dominating finite element among the
+     * last two elements of this collection (i.e., @p fes is `{2,3}`), then the
+     * answer is FE_Q(4) and therefore this function will return its index in
+     * the FECollection, namely `3`.
+     *
+     * This function differs from find_least_face_dominating_fe() in such a way
+     * that it only considers the subset @p fes, instead of finding a dominating
+     * element in the whole FECollection.
+     *
+     * For the purpose of this function by domination we consider either
+     * FiniteElementDomination::Domination::this_element_dominates or
+     * FiniteElementDomination::Domination::either_element_can_dominate;
+     * therefore the element can dominate itself. Thus, if an FECollection
+     * contains `{FE_Q(1),FE_Q(2),FE_Q(3),FE_Q(4)}` and @p fes only has
+     * a single element `{3}`, then the function returns 3.
+     *
+     * If the function is not able to find a finite element, the function
+     * returns numbers::invalid_unsigned_int .
+     *
+     * For an object, such as a line or a quad iterator, determine
+     * the fe_index of the most dominating finite element that
+     * lives on this object.
+     *
+     * Note that for the cases like when FECollection consists of `{FE_Nothing
+     * x FE_Nothing, FE_Q(1)xFE_Q(2), FE_Q(2)xFE_Q(1)}` with @p fes as `{1}`,
+     * the function will not find a most dominating element as the default
+     * behavior of FE_Nothing is to return
+     * FiniteElementDomination::no_requirements when comparing for face
+     * domination with any other element. This, therefore, can't be considered
+     * as a dominating element in the sense described above.
+     */
+    unsigned int
+    find_most_face_dominating_fe_in_subset(
```

Comment:
I think `most_` is not needed here. 

The intention of `least_` in the other function was to explain the fact that once we find all the dominating FE witihin the whole collection (we could have many), we choose the one which is dominated by everyone which will be the largest FE space and have least constraints.

whereas in your case you really look for dominating FE within a subset (i.e. square instead of rectangular). Assuming that we don't have funny cases where there are more than one solution, you don't need to choose `least_` or `most_`